lib/spack/spack/ci.py

```diff
@@ -1289,33 +1274,15 @@ def main_script_replacements(cmd):
                         ]
                     )
 
-                job_object = {
-                    "stage": stage_name,
-                    "variables": variables,
-                    "script": job_script,
-                    "tags": tags,
-                    "artifacts": {"paths": artifact_paths, "when": "always"},
-                    "needs": sorted(job_dependencies, key=lambda d: d["job"]),
```

Comment:
At the moment I don't recall why I made these sorted, but unless I missed something, I think we lose the sorting in this PR.  I can't think why it might matter, I'm just pointing it out.

Reply:
No, I think it's fine, thanks.